### PR TITLE
Aa fo connect to firestore

### LINKED ISSUE #1: 1. As a component, I want to be able to read from and write to the Firestore database so that users can persist information.
This is 1 of 2 stories that should be completed in the first week to enable further development of the app.

AC:

* The following have been added as project dependencies: `firebase` & `react-firebase-hooks`
* Make a change in the Firestore database and it shows up in the app
* Make a change in the app and it shows up in the Firestore database

Notes:

* Don't do too much here. In past, pairs have built the whole add item form in the first week. The best approach is just to make a throw away view with a `<button/>` and a `<div/>`. Click the `<button/>` to send some data to the database. Have the data show up in the `<div/>`. Simple!
* Also, if you need more direction on how to use `react-firebase-hooks`, you can reference this commit from Alejandro: https://github.com/alejandronanez/smart-shopping-list-demo/commit/d590adc3224872e71d000c41977a1b7a84542c31 and/or watch this Loom video where he explains how it works: https://www.loom.com/share/0dffa1a63bf24b449d6565c76606bbd9

### PR DESCRIPTION
**Pull request by fatima-ola**
## Description
This PR reads from and writes to the Firestore database so that users can persist information. In this PR, Ayu and Fatima created two components for the feature(AddItem and ListItem). Ayu worked on the ListItem component and was able to read from firestore while Fatima worked on the AddItem component to write to firestore. The code was then refactored and clean up. We learnt how to work together and collaborate effectively, we also learnt how to debug and fix errors by googling and asking questions.

### Write items to the database
We set up firestore on our codebase and connected it to the database so that we can write to it. We created a button and wrote a function for the button that helps submit our hard-coded data to firestore on a click action.

Check out more documentation and examples of how to write to firestore [here](https://firebase.google.com/docs/firestore/manage-data/add-data)

### Read items to the database
We set up firestore on our codebase and connected it to the database so that we can read from it. We populated our data from firestore to users immediately after the page loads. We make use of useeffect hooks to create a function that reads from firestore.

More documentation of effect hooks [here](https://reactjs.org/docs/hooks-effect.html)
Check out more documentation and examples of how to read from firestore [here](https://firebase.google.com/docs/firestore/query-data/get-data)

## Related Issue

closes #1 

## Acceptance Criteria

  ✓ The following have been added as project dependencies: firebase v9
  ✓Make a change in the Firestore database and it shows up in the app
  ✓Make a change in the app and it shows up in the Firestore database

## Type of Changes

|     | Type                       |
| --- | -------------------------- |
|  ✓| :bug: Bug fix              |
|  ✓ | :sparkles: New feature     |
|  ✓ | :hammer: Refactoring       |
|    | :100: Add tests            |
|   ✓ | :link: Update dependencies |
|    | :scroll: Docs              |

### After

collabpr

## Testing Steps / QA Criteria

- From your terminal, pull down this branch with git pull origin aa-fo-connect-to-firestore and check that branch out with git checkout aa-fo-connect-to-firestore
- Then npm update to install the newly added dependencies locally and npm start to launch the app.
- Onload, the data in firestore is populated and displayed on the page. 
- To test writing to firestore feature, navigate to AddItem.js file and input a new itemName and buyingTime, save your changes, and click Add Item button on the page to add the item. The item is added to firestore and displayed on the page.

### CODE REVIEW

File: src/components/ListItem.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+  const itemsCollectionRef = collection(db, 'shopping-list');
+
+  // get items
+  useEffect(() => {
+    const getItems = async () => {
+      const data = await getDocs(itemsCollectionRef);
+      setItems(data.docs.map((doc) => ({ ...doc.data(), id: doc.id })));
+    };
+    getItems();
+  }, []);
```
Comment (Amaka202) [inline src/components/ListItem.js:16]: This really works. The data is fetched but just once. When the user adds other items to the shopping list, they don't show up immediately except if the user refreshes the browser to get the newly added items. The firebase doc suggests that we add a listener to listen for changes in the firestore which will initiate another get request to fetch the newly added items. 

This is how I implemented it locally and it works really well. You don't need to refresh the browser to see new items added.

```
const ListItem = () => {
  const [items, setItems] = useState([]);
  const [loading, setLoading] = useState(false);
  const [error, setError] = useState(false);


  useEffect(() => {
    setLoading(true);
    const getItems = () => {
      try {
        const itemsCollectionRef = collection(db, 'shopping-list');
        const queryShoppingList = query(itemsCollectionRef);

        onSnapshot(queryShoppingList, (querySnapshot) => {
          const items = querySnapshot.docs.reduce((acc, doc) => {
            const { itemName, buyingTime } = doc.data();
            const id = doc.id;
            return [...acc, { id, itemName, buyingTime }];
          }, []);

          setItems(items);
          setLoading(false);
        });
      } catch (e) {
        setError(true);
      }
    };

    getItems();

  }, []);

  return (
    <div>
      {error && <p>An error occured while getting your items</p>}
      {loading && <p>Loading...</p>}
      {items.map((item) => {
        return (
          <div>
            <p>Name of Item: {item.itemName}</p>
            <p>Buying Time: {item.buyingTime}</p>
          </div>
        );
      })}
    </div>
  );
};

export default ListItem;
```

Hope this is helpful.👍👍

**Review by Amaka202 — CHANGES_REQUESTED** [review]
Nice job team!👍. I left a comment about how you can get data in real-time without refreshing the browser.

File: src/components/ListItem.js (hunk 1 above)
Comment (adiati98) [reply src/components/ListItem.js:16]: Hi @Amaka202,
Thank you for the feedback! 😀

I and @fatima-ola will take look and work on this, and testing it out too 😊

File: src/components/AddItem.js [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+        buyingTime: 'soon',
+      });
+
+      setLoading(false);
+    } catch (error) {
+      setError(true);
+      console.log('Error adding shopping-list:', error);
+    }
+  };
+
```
[the remark this answers is not in the document]
Comment (fatima-ola) [reply src/components/AddItem.js:25]: Thank you 🤗

**Comment by adiati98** [comment]
We have [updated the ListItem.js](https://github.com/the-collab-lab/tcl-35-smart-shopping-list/tree/aa-fo-connect-to-firestore) to include `onSnaphot` methods to listen to updated data in real-time.

**Review by Amaka202 — APPROVED** [review]
it works perfectly! well done team!!🎉🎉

**Pull request merged**